airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py

```diff
@@ -220,6 +220,11 @@ def process_status(
             # However, need to free the executor slot from the current executor.
             self.log.info("Event: pod %s adopted, annotations: %s", pod_name, annotations_string)
             self.watcher_queue.put((pod_name, namespace, ADOPTED, annotations, resource_version))
+        elif hasattr(pod.status, "reason") and pod.status.reason == "ProviderFailed":
+            self.log.error("Event: %s ProviderFailed, annotations: %s", pod_name, annotations_string)
```

Comment:
I think we should be a little more verbose in this error message. Just `ProviderFailed` isn't intuitive enough imo.